src/Notification/Job/SendNotificationsJob.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Flarum\Notification\Job;
+
+use Carbon\Carbon;
+use Flarum\Notification\Blueprint\BlueprintInterface;
+use Flarum\Notification\Event\Notifying;
+use Flarum\Notification\Event\Sending;
+use Flarum\Notification\MailableInterface;
+use Flarum\Notification\Notification;
+use Flarum\Notification\NotificationMailer;
+use Flarum\Notification\NotificationSyncer;
+use Flarum\Queue\AbstractJob;
+use Flarum\User\User;
+
+class SendNotificationsJob extends AbstractJob
+{
+    /**
+     * @var BlueprintInterface
+     */
+    private $blueprint;
+    /**
+     * @var array
+     */
+    private $recipientIds;
+
+    public function __construct(BlueprintInterface $blueprint, array $recipientIds = [])
+    {
+        $this->blueprint = $blueprint;
+        $this->recipientIds = $recipientIds;
+    }
+
+    public function handle(NotificationMailer $mailer)
+    {
+        $now = Carbon::now('utc')->toDateTimeString();
+        $recipients = $this->recipientIds;
+
+        event(new Sending($this->blueprint, $recipients));
+
+        $attributes = NotificationSyncer::getAttributes($this->blueprint);
+
+        Notification::insert(
+            array_map(function (User $user) use ($attributes, $now) {
+                return $attributes + [
+                        'user_id' => $user->id,
+                        'created_at' => $now
+                    ];
+            }, $recipients)
+        );
```

Comment:
Do we want the platform notifications to be generated here in the queue as well, or in sync? I could see arguments for both sides... :thinking:

Reply:
I considered both as well. But I think it would be smart to stick to one way of processing for all notification channels, unless we really have to make exceptions. Right now, pusher notifications can be sent async as well without issue.

Reply:
Fair enough. We should make sure the insertion of `Notifications` is idempotent, though - so that they are not re-created when a failure happens during mail delivery. Or is the email already sent via queue as well?